### PR TITLE
stlye: use thelper without linting every test

### PR DESCRIPTION
**Pull request by faddat**
## What is the purpose of the change

This PR introduces the thelper linter without moving to lint the tests.  It also includes a breaking 
change to an internal api, newSimulatorState() to make it more consistent the linter specifies that 
the testing library be passed as the first parameter. 

## Brief Changelog

* introduce thelper and begin each helper function outside of test files with a call to t.Helper() 
or tb.Helper()
* newSimulatorState now passes tb first, to be consistent
* changes a call from WithJSONCodec to WithCodec

## Testing and Verifying

This change enhanced tests by allowing them to give more feedback to developers in some situations. 

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? yes
  - How is the feature or change documented? (not applicable

### CODE REVIEW

**Comment by faddat** [comment]
changelog reminder ftw

**Comment by faddat** [comment]
labels: backport but only to v15

File: tests/e2e/containers/containers.go [hunk 1]
```diff
@@ -106,9 +109,11 @@ func (m *Manager) ExecHermesCmd(t *testing.T, command []string, success string)
 // returns container std out, container std err, and error if any.
 // An error is returned if the command fails to execute or if the success string is not found in the output.
 func (m *Manager) ExecCmd(t *testing.T, containerName string, command []string, success string) (bytes.Buffer, bytes.Buffer, error) {
+	t.Helper()
 	if _, ok := m.resources[containerName]; !ok {
 		return bytes.Buffer{}, bytes.Buffer{}, fmt.Errorf("no resource %s found", containerName)
 	}
+	t.Helper()
```
Comment (p0mvn) [inline tests/e2e/containers/containers.go:116]: duplicate
```suggestion
```

**Review by p0mvn — APPROVED** [review]

**Comment by faddat** [comment]
thank you for catching the dupe @p0mvn

**Review by czarcas7ic — APPROVED** [review]

**Comment by czarcas7ic** [comment]
Thanks for the addition @faddat, merging!

**Comment by faddat** [comment]
thanks :D it will make the linting of the tests way easier :)

**Comment by p0mvn** [comment]
Resolved conflict in the changelog. @faddat FYI, we shouldn't need a changelog entry for linter runs or updated.

Adding "no-changelog" label instead should be good.

This PR should be good to merge once CI passes. Thanks for the change @faddat

**Pull request merged**